example/src/main/java/com/stripe/example/view/CreditCardView.java

```diff
@@ -0,0 +1,560 @@
+                    }
+                }
+                return false;
+            }
+        });
+
+        mEtExpiryDate.addTextChangedListener(new TextWatcher() {
+            boolean freeze = false;
+            boolean hadSlash = false;
+
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int start, int count, int after) {
+                if (!freeze) {
+                    hadSlash = mEtExpiryDate.getText().toString().contains("/");
+                }
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int start, int before, int count) {
+                if (!freeze) {
+                    String curString = mEtExpiryDate.getText().toString();
+
+                    if (curString.isEmpty()) {
+                        return;
+                    }
+
+                    if (hadSlash && !mEtExpiryDate.getText().toString().contains("/")) {
+                        freeze = true;
+                        mEtExpiryDate.setText("");
+                        freeze = false;
+                        return;
+                    }
+
+                    mEtExpiryDate.setTextColor(mColor);
+
+                    String monthStr = null;
+                    String yearStr = null;
+
+                    String []parts = curString.split("[/]");
+                    if (parts.length > 0 && !StripeTextUtils.isBlank(parts[0])) {
+                        try {
+                            monthStr = parts[0].substring(0, parts[0].length() > 2 ? 2 : parts[0].length());
+                            int month = Integer.parseInt(monthStr);
+                            mCard.setExpMonth(month);
+                            if (parts[0].length() == 1 && month > 1) {
+                                monthStr = String.format("%02d", month);
+                            }
+                        } catch (NumberFormatException e) {
+                            e.printStackTrace();
+                            freeze = true;
+                            mEtExpiryDate.setText("");
+                            freeze = false;
+                            return;
+                        }
+                        if (mCard.validateExpMonth() && parts.length > 1) {
+                            try {
+                                yearStr = parts[1].substring(0, parts[1].length() > 2 ? 2 : parts[1].length());
+                                int year = Integer.parseInt(yearStr);
+                                mCard.setExpYear(year);
+                            } catch (NumberFormatException e) {
+                                e.printStackTrace();
+                                freeze = true;
+                                mEtExpiryDate.setText("");
+                                freeze = false;
+                                return;
+                            }
+                        }
+                    }
+
+                    freeze = true;
```

Comment:
The second situation would be ideal, but the only way to do that I can think of would be to unregister the TextWatcher and then register after the change is made. Using the freeze flag is not the prettiest but it does work and is not too costly.

I've refactored the variable name to ignoreTextChanges and reversed the boolean test to hopefully make it more readable.